run_ejb_app_tests.sh

```diff
@@ -1,27 +1,30 @@
 #!/usr/bin/env bash
 # Runs EJB App tests (ejb-core/rust/ejb-app).
+# If --skip-compile is passed, does not recompile all Java sources.
 #
 # ¡Keep it MacOS/Ubuntu compatible!
 
 # Fail immediately in case of errors and/or unset variables
 set -eu -o pipefail
 
-# Use the Java that Maven uses.
-#
-# Unfortunately, a simple `which java` will not work for some users (e.g., jenv),
-# hence this a bit complex thing.
-JAVA_HOME="$(mvn --version | grep 'Java home' | sed 's/.*: //')"
-echo "JAVA_HOME=${JAVA_HOME}"
+# Import necessary environment variables (see the tests_profile header comment for details).
+source tests_profile
 
-# Find the directory containing libjvm (the relative path has changed in Java 9)
-export LD_LIBRARY_PATH="$(find ${JAVA_HOME} -type f -name libjvm.* | xargs -n1 dirname)"
-echo "LD_LIBRARY_PATH=${LD_LIBRARY_PATH}"
+# Compile all Java modules by default to ensure that ejb-fakes module, which is required
+# by native ITs, is up-to-date. This safety net takes about a dozen seconds,
+# so if the Java artefacts are definitely up-to-date, it may be skipped.
+if [ "$#" -eq 0 ]; then
+  # Compile Java artefacts.
+  echo "Compiling the Java artefacts…"
+  mvn -DskipTests compile --quiet
```

Comment:
@vitvakatu , @alexander-irbis ,
Do ejb-app tests depend on any Java classes? If not, I believe this code is not needed.

Reply:
AFAIK not